kratos/utilities/entities_utilities.cpp

```diff
@@ -12,16 +12,201 @@
+    if constexpr (std::is_same<TEntity, Element>::value) {
+        entity_type_name = "Element";
+    } else if constexpr (std::is_same<TEntity, Condition>::value) {
+        entity_type_name = "Condition";
+    } else {
+        static_assert(std::is_same<TEntity, Element>::value || std::is_same<TEntity, Condition>::value, "Unsupported entity type. Only element and condition are supported.");
+    }
+    return entity_type_name;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+void EntitityIdentifier<TEntity>::GenerateSingleType(const std::string& rName)
 {
+    // Check the type of entity considered
+    const std::string entity_type_name = GetEntityTypeName();
+
+    // Some values need to be mandatorily prescribed since no meaningful default value exist. For this reason try accessing to them so that an error is thrown if they don't exist
+    KRATOS_ERROR_IF(rName != "" && !KratosComponents<TEntity>::Has(rName)) << entity_type_name << " name not found in KratosComponents<" << entity_type_name << "> -- name is " << rName << std::endl;
+    mpPrototypeEntity = KratosComponents<TEntity>::Get(rName).Create(0, nullptr, nullptr);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+void EntitityIdentifier<TEntity>::GenerateMultipleTypes(const std::string& rName)
+{
+    // Check the type of entity considered
+    const std::string entity_type_name = GetEntityTypeName();
+
+    // Split the name
+    const auto splitted_names = StringUtilities::SplitStringByDelimiter(rName, ';');
+    for (auto& r_entity_name : splitted_names) {
+        // Some values need to be mandatorily prescribed since no meaningful default value exist. For this reason try accessing to them so that an error is thrown if they don't exist
+        KRATOS_ERROR_IF(r_entity_name != "" && !KratosComponents<TEntity>::Has(r_entity_name)) << entity_type_name << " name not found in KratosComponents<" << entity_type_name << "> -- name is " << r_entity_name << std::endl;
+        const auto& r_ref_entity = KratosComponents<TEntity>::Get(r_entity_name);
+        const auto& r_reference_geometry = r_ref_entity.GetGeometry();
+        const auto& r_reference_geometry_type = r_reference_geometry.GetGeometryType();
+        mTypes.insert({r_reference_geometry_type, &KratosComponents<TEntity>::Get(r_entity_name)});
+    }
+
+    // Set the definition type
+    mDefinitionType = EntitiesUtilities::DefinitionType::Multiple;
+
+    // Assign dumy prototype
+    typename GeometryType::Pointer p_geometry_type = Kratos::make_shared<GeometryType>();
+    mpPrototypeEntity = KratosComponents<TEntity>::Get(entity_type_name + "2D2N").Create(0, p_geometry_type, nullptr);
```

Comment:
i don't understadn what you do here... why do you generate a type 2D2N?

if it is something you cannot use "as is" i would leave it a nullptr

Reply:
well, to be honesty i don't think this is a good solution. 

the entity type is a nullptr up to the point at which you have the correct entity... 

both p_geometry_type and mpPrototypeEntity should be either nullptrs or have the correct type

Reply:
Fixed